### DISCUSSION TITLE
Handle missmatching spectra

### DISCUSSION THREAD

**Discussion by sschusser** [spectrochempy/spectrochempy#644] [Q&A] [answered]
Is there any way to handle two spectra of different resolution and/or bounds?

I know, it is not possible to beat physics. So, I do not expect any reults for the region outside the given datapoints. Nevertheless, it should be possible to do some math on two spectra of different wavenumbers in the table.

**Comment by fernandezc** [comment]
It depends on what exactly you want to do. It seems impossible to do math such as addition, multiplication... etc, on spectra with different coordinates. But it is always possible to align them before doing calculations. This means that some of the spectra must be transformed. If this is not what you want, could you give an example (maybe some code to start with)

**Reply by fernandezc** [reply]
In addition, SpectroChemPy has already some tools to align spectra, but one thing is missing: it is when it is necessary to interpolate between points.  But this can be quite easily solved in a next future.

**Reply by sschusser** [reply]
I have to manage two different kinds of missmatch:

1. Missmatch of wavenumber sampling:
In general, the spectrum covers the same range but the number of datapoints differ. Thus, i am thinking of a kind of resampling to get corresponding values. In most cases the spectra are oversampled. Hence, I do not expect information loss or significant distortion of the spectra.

2. Different range:
I often come across spectra of varying spectral range. Hence, two spectra do not overlap completely. Managing this missmatch is tricky because of the missing information about the missing spectral region. Thus, I expect that it becomes necessary to choose from several rules how to deal with the missing information. I am thinking of some basic strategies, such as "fill with value" or "hold last value".


I am new to this topic and was hoping that smart people solved such basic stumbling blocks already. I expected that such kind of missmatchs are annoying but common business handling spectroscopic data - some times from different sources. So far I could not find any promissing information on how to manage this issues with spectrochempy. Any hints? You mentioned to align and transform spectra.

**Reply by fernandezc** [reply, marked as answer]
Align: https://www.spectrochempy.fr/latest/userguide/processing/alignment.html 

This could solve some problems. 

For resampling (and interpolation), it is possible to do this even if SpectroChemPy does not yet propose all the tools. SciPy and numpy may be useful for manipulating the data in addition to what SpectroChemPy already offers. I will try to write a working example when I have time. 

Also, if you have some examples of misaligned spectra that you would like to address, perhaps you can send them to us by attaching them here.
